### PR TITLE
[SD-4591]-(fix) - Integrated editor2 to aap branch

### PR DESCRIPTION
**Pull request by sjunaid**
- port text expansion updates in editor2
- in-cooperated editor config settings
- fixed improper find-replace all

### CODE REVIEW

File: scripts/superdesk/editor2/editor.js [hunk 1]
```diff
@@ -290,7 +311,7 @@ function EditorService(spellcheck, $q, _, renditionsService, utils) {
         for (var i = 0, l = nodes.length; i < l; i++) {
             var node = nodes.item(i);
             var word = node.dataset.word;
-            var index = parseInt(node.dataset.index, 10);
+            var index = scope.node.textContent.indexOf(word);
```
Comment (petrjasek) [inline scripts/superdesk/editor2/editor.js]: this should use node, otherwise it ignores filtering done in `replaceText`

**Comment by akintolga** [comment]
:+1:

**Comment by marwoodandrew** [comment]
:+1:

File: scripts/superdesk/editor2/views/block-text.html [hunk 2]
```diff
@@ -13,7 +13,8 @@
         data-append-class="active"
         placeholder="{{ ::'Add content here' | translate }}"
         ng-show="type=='text' || !type"
-        spellcheck="false">
+        spellcheck="false"
+        ng-attr-data-disable-toolbar="{{(!toolbarConfig || config.disableToolbar) || undefined}}">
```
Comment (petrjasek) [inline scripts/superdesk/editor2/views/block-text.html]: just wondering - what is this doing?

Comment (sjunaid) [inline scripts/superdesk/editor2/views/block-text.html]: @petrjasek , removed seems not necessary as discussed.

**Pull request merged**